### ISSUE TITLE
I've found some bugs.

### ISSUE THREAD

**Issue by wcshin-git** [lucidrains/DALLE-pytorch#265]
Hi there, 
I read the training code line by line, and I've found some bugs.
* When making the contextualized vector representations of the image tokens, the model references text pad tokens. I think you need to mask the text pad parts for that!
* When making the text loss, you included the loss from the text pad tokens. I think it should be removed from the text loss. 

Thanks for updating the code!

**Comment by lucidrains** [comment]
@wcshin-git Hi Woncheol! Glad you have been studying the code :)

So, while you are correct that is the conventional way of doing things, if you read the paper, they actually do not mask out attention to the padding tokens. The paper takes a completely new turn, where they give each position a unique padding token. They neither exclude it from the loss nor mask out attention to it, as far as I understand it.

Hope this help!

**Comment by lucidrains** [comment]
<img width="351" alt="Screen Shot 2021-05-29 at 10 24 57 AM" src="https://user-images.githubusercontent.com/108653/120079329-2aac5180-c068-11eb-9df0-ae977c485fb1.png">
